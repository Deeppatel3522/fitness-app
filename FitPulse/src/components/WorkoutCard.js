import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, Image } from 'react-native';
import { getExerciseImage } from '../constants/exerciseImages';
import { Ionicons } from '@expo/vector-icons';

const WorkoutCard = ({ workout, onPress }) => {
  const getDifficultyColor = (difficulty) => {
    switch(difficulty) {
      case 'Beginner': return '#4CAF50';
      case 'Intermediate': return '#FF9800';
      case 'Advanced': return '#F44336';
      default: return '#2196F3';
    }
  };

  // Get a preview image (you can use the first exercise of the workout)
  const previewImage = workout.previewExercise ? 
    getExerciseImage(workout.previewExercise) : 
    require('../../assets/images/exercises/default-exercise.png');

  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Image 
        source={previewImage}
        style={styles.previewImage}
        resizeMode="cover"
      />
      <View style={styles.content}>
        <Text style={styles.name}>{workout.name}</Text>
        <View style={styles.details}>
          <Text style={styles.duration}>{workout.duration}</Text>
          <View style={[
            styles.difficultyBadge,
            { backgroundColor: getDifficultyColor(workout.difficulty) }
          ]}>
            <Text style={styles.difficultyText}>{workout.difficulty}</Text>
          </View>
        </View>
      </View>
      <View style={styles.arrow}>
        <Ionicons name='play' size={16} color={"#4CAF50"}/>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#f8f8f8',
    padding: 15,
    borderRadius: 10,
    marginBottom: 10,
  },
  previewImage: {
    width: 60,
    height: 60,
    borderRadius: 10,
    marginRight: 15,
  },
  content: {
    flex: 1,
  },
  name: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 8,
  },
  details: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  duration: {
    fontSize: 14,
    color: '#666',
    marginRight: 10,
  },
  difficultyBadge: {
    paddingHorizontal: 8,
    paddingVertical: 3,
    borderRadius: 10,
  },
  difficultyText: {
    fontSize: 12,
    color: '#fff',
    fontWeight: 'bold',
  },
  arrow: {
    marginLeft: 10,
    paddingRight: 25
  },
  arrowText: {
    fontSize: 16,
    color: '#4CAF50',
  },
});

export default WorkoutCard;
