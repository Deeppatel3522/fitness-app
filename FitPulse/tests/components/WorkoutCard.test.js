import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import WorkoutCard from '../../src/components/WorkoutCard';

// Mock getExerciseImage to avoid image loading issues
jest.mock('../../src/constants/exerciseImages', () => ({
  getExerciseImage: jest.fn(() => ({ uri: 'mock-exercise-image.jpg' })),
}));

// Mock react-native-image-placeholder
jest.mock('react-native-image-placeholder', () => 'MockedImageLoad');

describe('WorkoutCard', () => {
  const mockWorkout = {
    id: '1',
    name: 'Morning Cardio',
    duration: '30 min',
    difficulty: 'Intermediate',
    previewExercise: 'Push-ups',
  };

  const mockOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders workout information correctly', () => {
    const { getByText, getByTestId } = render(
      <WorkoutCard workout={mockWorkout} onPress={mockOnPress} />
    );

    expect(getByText('Morning Cardio')).toBeTruthy();
    expect(getByText('30 min')).toBeTruthy();
    expect(getByText('Intermediate')).toBeTruthy();
    expect(getByTestId('workout-card')).toBeTruthy();
  });

  it('calls onPress when card is tapped', () => {
    const { getByTestId } = render(
      <WorkoutCard workout={mockWorkout} onPress={mockOnPress} />
    );

    const card = getByTestId('workout-card');
    fireEvent.press(card);

    // âœ… Expect onPress to be called once, without any arguments
    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(mockOnPress).toHaveBeenCalledWith(); // No arguments expected
  });

  it('renders different difficulty levels correctly', () => {
    const difficulties = ['Beginner', 'Intermediate', 'Advanced'];
    
    difficulties.forEach(difficulty => {
      const testWorkout = { ...mockWorkout, difficulty };
      const { getByText } = render(
        <WorkoutCard workout={testWorkout} onPress={mockOnPress} />
      );
      
      expect(getByText(difficulty)).toBeTruthy();
    });
  });

  it('handles workout without previewExercise', () => {
    const workoutWithoutPreview = {
      ...mockWorkout,
      previewExercise: undefined,
    };

    const { getByText } = render(
      <WorkoutCard workout={workoutWithoutPreview} onPress={mockOnPress} />
    );

    expect(getByText('Morning Cardio')).toBeTruthy();
    expect(getByText('30 min')).toBeTruthy();
    expect(getByText('Intermediate')).toBeTruthy();
  });

  it('renders with custom difficulty colors', () => {
    const beginnerWorkout = { ...mockWorkout, difficulty: 'Beginner' };
    const advancedWorkout = { ...mockWorkout, difficulty: 'Advanced' };

    const { getByText: getByTextBeginner } = render(
      <WorkoutCard workout={beginnerWorkout} onPress={mockOnPress} />
    );
    
    const { getByText: getByTextAdvanced } = render(
      <WorkoutCard workout={advancedWorkout} onPress={mockOnPress} />
    );

    expect(getByTextBeginner('Beginner')).toBeTruthy();
    expect(getByTextAdvanced('Advanced')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(
      <WorkoutCard workout={mockWorkout} onPress={mockOnPress} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
