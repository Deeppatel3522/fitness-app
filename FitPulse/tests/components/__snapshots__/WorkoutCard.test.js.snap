// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutCard matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f8f8f8",
      "borderRadius": 10,
      "flexDirection": "row",
      "marginBottom": 10,
      "opacity": 1,
      "padding": 15,
    }
  }
  testID="workout-card"
>
  <Image
    resizeMode="cover"
    source={
      {
        "uri": "mock-exercise-image.jpg",
      }
    }
    style={
      {
        "borderRadius": 10,
        "height": 60,
        "marginRight": 15,
        "width": 60,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      Morning Cardio
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 14,
            "marginRight": 10,
          }
        }
      >
        30 min
      </Text>
      <View
        style={
          [
            {
              "borderRadius": 10,
              "paddingHorizontal": 8,
              "paddingVertical": 3,
            },
            {
              "backgroundColor": "#FF9800",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 12,
              "fontWeight": "bold",
            }
          }
        >
          Intermediate
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginLeft": 10,
        "paddingRight": 25,
      }
    }
  >
    <MockedIonicons
      color="#4CAF50"
      name="play"
      size={16}
    />
  </View>
</View>
`;
